src/Net/SocketExtensions.cs

```diff
@@ -22,9 +22,32 @@ namespace Amqp
     using System.Net;
     using System.Net.Sockets;
     using System.Threading.Tasks;
+    using System.Runtime.InteropServices;
 
     static class SocketExtensions
     {
+        public static void SetTcpKeepAlive(this Socket socket, TcpKeepAliveSettings settings)
+        {
+            /* the native structure
+            struct tcp_keepalive {
+            ULONG onoff;
+            ULONG keepalivetime;
+            ULONG keepaliveinterval;
+            };
+            */
+
+            ulong keepAliveTime = settings.KeepAliveTime;
+            ulong keepAliveInterval = settings.KeepAliveInterval;
+
+            int bytesPerULong = 8;
+            byte[] inOptionValues = new byte[bytesPerULong * 3];
+            BitConverter.GetBytes((ulong)1).CopyTo(inOptionValues, 1);
```

Comment:
Just curious, why copy to destination array at index 1?

Reply:
I was just looking at it wrong.. reverted this in a following commit